tensorflow_recommenders_addons/dynamic_embedding/core/kernels/lookup_impl/lookup_table_op_hkv.h

```diff
@@ -0,0 +1,712 @@
+                      cudaStream_t stream, const size_t buffer_size) {
+    std::unique_ptr<nv::merlin::BaseKVFile<K, V, uint64_t>> rfile;
+    string keyfile = filepath + "-keys";
+    string valuefile = filepath + "-values";
+    string metafile = filepath + "-metas";
+    bool has_metas = false;
+    Status status = Status::OK();
+
+    if (is_valid_metas(keyfile, metafile)) {
+      rfile.reset(new nv::merlin::LocalKVFile<K, V, uint64_t>);
+      bool open_ok = reinterpret_cast<nv::merlin::LocalKVFile<K, V, uint64_t>*>(
+                         rfile.get())
+                         ->open(keyfile, valuefile, metafile, "rb");
+      has_metas = true;
+      if (!open_ok) {
+        std::string error_msg = "Failed to load from file " + keyfile + ", " +
+                                valuefile + ", " + metafile;
+        throw std::runtime_error(error_msg);
+      }
+    } else {
+      rfile.reset(
+          new RandomKVFile<K, V, uint64_t>(fs, filepath, dim, buffer_size));
+      status = reinterpret_cast<RandomKVFile<K, V, uint64_t>*>(rfile.get())
+                   ->open(keyfile, valuefile, "rb");
+    }
+    if (!status.ok()) {
+      std::string error_msg = "Failed to load from file " + keyfile + ", " +
+                              valuefile + ", " + metafile + " " +
+                              status.ToString();
+      throw std::runtime_error(error_msg);
+    }
+
+    size_t n_loaded = table_->load(rfile.get(), buffer_size, stream);
+    if (has_metas) {
+      LOG(INFO) << "[op] Load " << n_loaded
+                << " pairs from keyfile: " << keyfile
+                << ", and valuefile: " << valuefile << ", and metafile"
+                << metafile;
+    } else {
+      LOG(INFO) << "[op] Load " << n_loaded
+                << " pairs from keyfile: " << keyfile
+                << ", and valuefile: " << valuefile;
+    }
+    CUDA_CHECK(cudaStreamSynchronize(stream));
+    if (has_metas) {
+      reinterpret_cast<nv::merlin::LocalKVFile<K, V, uint64_t>*>(rfile.get())
+          ->close();
+    } else {
+      reinterpret_cast<RandomKVFile<K, V, uint64_t>*>(rfile.get())->close();
+    }
+  }
+
+  void get(const K* d_keys, V* d_vals, bool* d_status, size_t len, V* d_def_val,
+           cudaStream_t stream, bool is_full_size_default) const {
+    if (is_full_size_default) {
+      CUDA_CHECK(cudaMemcpyAsync(d_vals, d_def_val, sizeof(V) * dim_ * len,
+                                 cudaMemcpyDeviceToDevice, stream));
+    } else {
+      size_t grid_size = nv::merlin::SAFE_GET_GRID_SIZE(len, block_size_);
+      gpu_fill_default_values<V>
+          <<<grid_size, block_size_, dim_ * sizeof(V), stream>>>(
+              d_vals, d_def_val, len, dim_);
+    }
+    table_->find(len, d_keys, d_vals, d_status, /*d_metas=*/nullptr, stream);
+  }
+
+  // TODO: Implement a contain kernel instead of find.
+  void contains(const K* d_keys, V* d_status, size_t len, cudaStream_t stream) {
```

Comment:
The current master has implemented a `contains` API.

Reply:
Really? I didn't see it. There is also no place where this function is called. Can it be deleted?

Reply:
Sorry, I made a mistake. I was looking at the code for beta6. Additionally, I mistakenly thought that "master" referred to the master branch of tfra. Please ignore me.

Reply:
DONE